### PR TITLE
Changing packages structure, cleaning up JPA entities

### PR DESCRIPTION
**Pull request by BerezinD**

### CODE REVIEW

File: api/src/main/java/org/cardanofoundation/rosetta/common/services/ProtocolParamServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+@Slf4j
+@RequiredArgsConstructor
+public class ProtocolParamServiceImpl implements ProtocolParamService {
+
+  private final EpochParamRepository epochParamRepository;
+  private final ProtocolParamsToEntity mapperProtocolParams;
+
+  private ProtocolParams cachedProtocolParams;
+
+  @PostConstruct
```
Comment (Kammerlo) [inline api/src/main/java/org/cardanofoundation/rosetta/common/services/ProtocolParamServiceImpl.java:25]: This PostConstruct leads to a crash of the intregrationtest. I had the same problem with the genesisblock caching. 
A work-around would be to save not on postconstruct, but when it is queried the first time.

**Review by Kammerlo — APPROVED** [review]

File: api/src/main/java/org/cardanofoundation/rosetta/common/services/ProtocolParamServiceImpl.java (hunk 1 above)
Comment (BerezinD) [reply api/src/main/java/org/cardanofoundation/rosetta/common/services/ProtocolParamServiceImpl.java]: Thank you for the information!
I've added some old-style thread-safe code to get parameters on the first call:
https://github.com/cardano-foundation/cardano-rosetta-java/pull/169/files#diff-e1ab139173b73f29bec8b97daa5b2b6d8a04b47a287be793fa90672656284344R25

File: api/src/main/java/org/cardanofoundation/rosetta/common/mapper/DataMapper.java [hunk 2]
```diff
@@ -52,20 +50,23 @@
 public class DataMapper {
```
Comment (Gennadii-T) [inline api/src/main/java/org/cardanofoundation/rosetta/common/mapper/DataMapper.java:50]: Could you put it in separate methods Currency.builder() and Amount.builder() to maintain readability and avoid code duplication

**Review by Gennadii-T — APPROVED** [review]

**Review by AlksndrSlgb — APPROVED** [review]

File: api/src/main/java/org/cardanofoundation/rosetta/common/mapper/DataMapper.java (hunk 2 above)
Comment (BerezinD) [reply api/src/main/java/org/cardanofoundation/rosetta/common/mapper/DataMapper.java:50]: Do you mean moving the default currency to a separate method?
`Currency.builder()
        .decimals(Constants.ADA_DECIMALS)
        .symbol(Constants.ADA).build();`

**Review by shleger — APPROVED** [review]

File: api/src/main/java/org/cardanofoundation/rosetta/common/mapper/DataMapper.java (hunk 2 above)
Comment (BerezinD) [reply api/src/main/java/org/cardanofoundation/rosetta/common/mapper/DataMapper.java:50]: I will refactor it with the next PR, thanks for the comment.

**Pull request merged**